.github/workflows/entropy-check.yml

```diff
@@ -0,0 +1,43 @@
+---
+name: entropy-check
+
+on:
+  pull_request_target:
+    branches:
+      - main
+  workflow_dispatch:
+
+
+jobs:
+  entropy-check:
+    runs-on: ubuntu-latest
+    permissions: write-all
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install python env
+        uses: ./.github/actions/install-python-env
+
+      - name: Process repository entropy
+        run: poetry run almanack process_repo_entropy . > entropy_report.txt
+
+      - name: Format Entropy Report
+        id: format_report
+        run: |
+          {
+            echo "REPORT_CONTENT<<EOF";
+            echo "\`\`\`";
+            cat entropy_report.txt;
+            echo "\`\`\`";
+            echo "EOF";
+          } >> "$GITHUB_ENV"
+
+      - name: PR comment
+        uses: thollander/actions-comment-pull-request@v2.5.0
```

Comment:
Consider using `@v2` to help constrain this by major version (leaving it open to improvements as they are released through the GH action).